src/template.rs

```diff
@@ -112,12 +112,115 @@ pub mod django_rusty_templates {
+                return Self::map_loader(
+                    py,
+                    &loader_str.extract::<String>()?,
+                    Vec::new(),
+                    HashMap::new(),
+                );
+            }
+
+            // Try as tuple (loader, args)
+            if let Ok(loader_tuple) = ld.downcast::<PyTuple>() {
+                if let Ok((loader_obj, args)) =
+                    loader_tuple.extract::<(Bound<'_, PyString>, Bound<'_, PyAny>)>()
+                {
+                    let loader_path = loader_obj.extract::<String>()?;
+
+                    // Process args based on type
+                    // Check if args is PyList
+                    if let Ok(true) = args.is_instance(&py.get_type::<PyList>()) {
+                        let args_list = args.downcast::<PyList>()?;
+                        let args_vec = args_list
+                            .iter()
+                            .map(|item| item.extract::<String>())
+                            .collect::<Result<Vec<String>, PyErr>>()?;
+
+                        return Self::map_loader(py, &loader_path, args_vec, HashMap::new());
+                    // Check if args is PyTuple
+                    } else if let Ok(true) = args.is_instance(&py.get_type::<PyTuple>()) {
+                        let args_tuple = args.downcast::<PyTuple>()?;
+                        let args_vec = args_tuple
+                            .iter()
+                            .map(|item| item.extract::<String>())
+                            .collect::<Result<Vec<String>, PyErr>>()?;
+
+                        return Self::map_loader(py, &loader_path, args_vec, HashMap::new());
+                    // Check if args is PyDict
+                    } else if let Ok(true) = args.is_instance(&py.get_type::<PyDict>()) {
+                        let args_dict = args.downcast::<PyDict>()?;
+                        let args_hashmap = args_dict
+                            .iter()
+                            .map(|(key, value)| {
+                                Ok((key.extract::<String>()?, value.extract::<String>()?))
+                            })
+                            .collect::<Result<HashMap<String, String>, PyErr>>()?;
+
+                        return Self::map_loader(py, &loader_path, Vec::new(), args_hashmap);
+                    } else {
+                        return Err(PyErr::new::<pyo3::exceptions::PyTypeError, _>(format!(
+                            "Unsupported args type: {:?}",
+                            args
+                        )));
+                    }
+                }
+            }
+
+            Err(PyErr::new::<pyo3::exceptions::PyTypeError, _>(format!(
+                "Unsupported loader type: {:?}",
+                ld
+            )))
+        }
+
+        fn map_loader(
+            py: Python<'_>,
+            loader_path: &str,
+            args_vec: Vec<String>,
+            args_hashmap: HashMap<String, String>,
```

Comment:
With the `unpack` refactor:

```suggestion
            args: Vec<Bound<'_, PyAny>>,
```